File: gosrc/leason3/cat.go
package leason3

import (
	"fmt"
)

type cat struct {
	name string
}

func NewCat(name string) Animal {
	ret := &cat{name}
	return ret
}

func (c *cat) Eat() {
	fmt.Println("fish")
}

func (c *cat) Name() string {
	return c.name
}

File: src/com/heiwawu/example/leason3/impl/Cat.java
package com.heiwawu.example.leason3.impl;

import com.heiwawu.example.leason3.Animal;

public class Cat implements Animal{
    private String name;

    public Cat(String name) {
        this.name = name;
    }

    @Override
    public void eat() {
        System.out.println("fish");
    }

    @Override
    public String name() {
        return this.name;
    }
}
